riscv_private.h

```diff
@@ -210,3 +238,67 @@ static inline uint32_t sign_extend_b(uint32_t x)
 {
     return (int32_t)((int8_t) x);
 }
+
+// decode rs1 field
+static inline uint16_t c_dec_rs1(uint16_t x){
+    return (uint16_t)((x & FC_RS1) >> 7U);
+}
+
+// decode rs2 field
+static inline uint16_t c_dec_rs2(uint16_t x){
+    return (uint16_t)((x & FC_RS2) >> 2U);
+}
+
+// decode rd field
+static inline uint16_t c_dec_rd(uint16_t x){
+    return (uint16_t)((x & FC_RD) >> 7U);
+}
+
+// decode rs1' field
+static inline uint16_t c_dec_rs1c(uint16_t x){
+    return (uint16_t)((x & FC_RS1C) >> 7U);
+}
+
+// decode rs2' field
+static inline uint16_t c_dec_rs2c(uint16_t x){
+    return (uint16_t)((x & FC_RS2C) >> 2U);
+}
+
+// decode rd' field
+static inline uint16_t c_dec_rdc(uint16_t x){
+    return (uint16_t)((x & FC_RDC) >> 2U);
+}
+
+static inline uint16_t c_dec_cjtype_imm(uint16_t x){
+    uint16_t temp = 0;
+    //                ....xxxx....xxxx
+    temp |= (x & 0b0000000000111000) >> 2;
+    temp |= (x & 0b0000100000000000) >> 7;
+    temp |= (x & 0b0000000000000100) << 3;
+    temp |= (x & 0b0000000010000000) >> 1;
+    temp |= (x & 0b0000000001000000) << 1;
+    temp |= (x & 0b0000011000000000) >> 1;
+    temp |= (x & 0b0000000100000000) << 2;
+    temp |= (x & 0b0001000000000000) >> 1;
+    // extend to 16 bit
+    for(int i = 1; i < 4; ++i){
+        temp |= (0x0800 & temp) << i;
+    }
+
+    return temp;
+}
+
+static inline uint16_t c_dec_cbtype_imm(uint16_t x){
+    uint16_t temp = 0;
+    //             ....xxxx....xxxx
+    temp |= (x & 0b0000000000011000) >> 2;
+    temp |= (x & 0b0000110000000000) >> 7;
+    temp |= (x & 0b0000000000000100) << 3;
+    temp |= (x & 0b0000000001100000) << 1;
+    temp |= (x & 0b0001000000000000) >> 4;
+    // extend to 16 bit
+    for(int i = 1; i < 8; ++i){
```

Comment:
Use `clang-format -i` to ensure the consistent styie.

Reply:
It's my negligence, I'm sorry.